src/rgw/rgw_s3select.cc

```diff
@@ -0,0 +1,656 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include "rgw_s3select.h"
+
+RGWOp* create_s3select_op()
```

Comment:
we could do this later, but let's put all the code in namespace rgw (or maybe rgw::s3select)? that would allow seeing more easily from rgw_rest_s3.cc what provides create_s3select_op()

Reply:
I do think the new rgw_s3select.h/cc file is a good cleanup

Reply:
agree, we should add a namespace.

Reply:
looks good